### ISSUE TITLE
maven-publish not parallelising subprojects

### ISSUE THREAD

**Issue by samcgardner** [gradle/gradle#23484]
### Expected Behavior
Publish tasks should be execute in parallel

### Current Behavior
Publish tasks are executed sequentially

### Context
Some publish targets (e.g. GitHub packages) are _incredibly_ slow, so for projects with lots of modules it's strongly desirable to parallelise these uploads to finish publishing in a reasonable time window.

### Steps to Reproduce 
```kotlin
publishing {
  // list of repositories to deploy to
  repositories {
    mavenLocal()
    maven {
      name = "GitHubPackages"
      url = URI.create("https://maven.pkg.github.com/${ORGANISATION}/${REPO}")
      credentials {
        username = project.findProperty("github.user") as String? ?: System.getenv("GITHUB_USERNAME")
        password = project.findProperty("github.token") as String? ?: System.getenv("GITHUB_TOKEN")
      }
    }
  }

  projectsToPublish.forEach { subProject ->
    subProject.apply { plugin<MavenPublishPlugin>() }

    publications {
      create("${subProject.name}-maven", MavenPublication::class) {
        this.groupId = subProject.group.toString()
        this.artifactId = subProject.name
        this.version = subProject.version.toString()

        this.from(subProject.components["java"])
        println("Published group ID: ${this.groupId} artifact ID: ${this.artifactId} version: ${this.version}")
      }
    }
  }
}
```
Create a project with lots of subprojects and configure them like so.

### Your Environment
Build scan URL: https://scans.gradle.com/s/7hvinhcygfcj4/

**Comment by ljacomet** [comment]
Thank you for your interest in Gradle!

Your issue appears to be a question about how to use Gradle. Your question might be answered in Gradle documentation or on our community Slack or forums. Find links to these resources on [help.gradle.org](https://help.gradle.org/).

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

If you think Gradle is missing a feature, please file a new issue describing your use case for the feature.

***

You are creating all the publications in the root project instead of inside each project. `parallel` does not work between tasks of a single project.
I would recommend using a [convention plugin](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins) to define your publications in each project that needs to be published.
